catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -452,60 +491,144 @@ private void moveToFailedIngestDirectory(File source) {
+
+                            if (ignoreList != null && (ignoreList.contains(extension)
+                                    || ignoreList.contains(file.getName()))) {
+                                ignoreCount.incrementAndGet();
+                                printProgressAndFlush(start,
+                                        fileCount.get(),
+                                        ingestCount.get() + ignoreCount.get());
+                            } else {
+                                Metacard result;
+                                try {
+                                    result = readMetacard(file);
+                                } catch (IngestException e) {
+                                    result = null;
+                                    logIngestException(e, file);
+                                    if (failedIngestDirectory != null) {
+                                        moveToFailedIngestDirectory(file);
+                                    }
+                                    printErrorMessage(
+                                            "Failed to ingest file [" + file.getAbsolutePath()
+                                                    + "].");
+                                    if (INGEST_LOGGER.isWarnEnabled()) {
+                                        INGEST_LOGGER.warn("Failed to ingest file [{}].",
+                                                file.getAbsolutePath());
+                                    }
                                 }
-                            }
 
-                            if (result != null) {
-                                try {
-                                    metacardQueue.put(result);
-                                } catch (InterruptedException e) {
-                                    INGEST_LOGGER.error(
-                                            "Thread interrupted while waiting to 'put' metacard: {}",
-                                            e);
+                                if (result != null) {
+                                    try {
+                                        metacardQueue.put(result);
+                                    } catch (InterruptedException e) {
+                                        INGEST_LOGGER.error(
+                                                "Thread interrupted while waiting to 'put' metacard: {}",
+                                                result.getId(),
+                                                e);
+                                    }
                                 }
                             }
                         }
-                    }
-                });
+                    });
+        }
         doneBuildingQueue.set(true);
     }
 
+    private void submitToStorageProvider(List<Metacard> metacardList) {
+        StorageProvider storageProvider = getAllServices(StorageProvider.class).get(0);
+
+        metacardList.forEach(metacard -> {
```

Comment:
You might want to consider refactoring this to something similar to:
```
metacardList.stream().filter(metacard -> metacardFileMapping.containsKey(metacard.getId()))
    .map(metacard -> {
        ...
        return new CreateStorageRequestImpl(...);
    }).forEach(createStorageRequest -> {
    ...
    });
```
I think it makes the code's intent a bit clearer.